protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -142,17 +139,17 @@ impl<'a, T: Serialize + GetSize + Deserialize<'a>, B: IsBuffer + AeadBuffer> Wit
                 }
                 self.sv2_buffer.danger_set_start(0);
                 let src = self.sv2_buffer.get_data_owned();
-                let frame = Sv2Frame::<T, B::Slice>::from_bytes_unchecked(src);
+                let frame = Sv2Frame::<T, B::Slice>::from_bytes(src)?;
                 Ok(frame.into())
             }
         }
     }
 
-    fn while_handshaking(&mut self) -> EitherFrame<T, B::Slice> {
+    fn while_handshaking(&mut self) -> Frame<T, B::Slice> {
         let src = self.noise_buffer.get_data_owned().as_mut().to_vec();
 
         // below is inffalible as noise frame length has been already checked
-        let frame = HandShakeFrame::from_bytes_unchecked(src.into());
+        let frame = HandShakeFrame::from_bytes(src.into());
```

Comment:
we can use unchecked here, if you are concerned you can put a check in debug mode